### PR TITLE
Issue #2290 Add minishift architecture diagram as image

### LINKED ISSUE #2290: Rendering of Minishift architecture image is broken
Minishift architecture image rendering is broken in the published docs [1].

[1] https://docs.openshift.org/latest/minishift/using/basic-usage.html

### PR DESCRIPTION
**Pull request by budhrg**
Fix #2290 

Here, we are generating locally and using the image in the `basic-usage.adoc` file.

Also tested locally by generating the adoc.tar file.

### CODE REVIEW

**Comment by budhrg** [comment]
retest this please

File: docs/Rakefile [hunk 1]
```diff
@@ -80,6 +80,7 @@ task :serve, [:server_args] => [:init, :markdown_to_asciidoc, :adoc_variables] d
   if args[:server_args].nil? then
     sh "bundle exec middleman server"
   else
+    sh "mv -f source/using/images/minishift-architecture-template.png source/using/images/minishift-architecture.png"
```
Comment (praveenkumar) [inline docs/Rakefile]: @budhrg If I am not wrong then if the image file not generated by any reason then even this `mv -f` also going to fail right? how we are sure that this is something going to work every time just changing the name of the file from `*.png.cache` to `*-template.png.cache`?

Comment (budhrg) [reply docs/Rakefile]: @praveenkumar 

> If I am not wrong then if the image file not generated by any reason then even this `mv -f` also going to fail right? 

Yes, if generation is failed than `mv -f` is going to fail.

> how we are sure that this is something going to work every time just changing the name of the file from *.png.cache to *-template.png.cache?

I didn't that about this. Here we are renaming the template image name to actual image name which will be used in `basic-usage.adoc` file.

Comment (LalatenduMohanty) [reply docs/Rakefile]: @budhrg Can you please point me to the code which generates `minishift-architecture-template.png`

**Comment by budhrg** [comment]
@LalatenduMohanty @praveenkumar Done with my changes. It will pass now (also run unit/integration test unnecessarily :wink: ) .

**Comment by budhrg** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
@praveenkumar WDYT?

**Comment by budhrg** [comment]
@LalatenduMohanty @praveenkumar I would suggest to merge this PR as an independent entity and the suggestion Praveen is giving is based on the assumption that if the image get failed in generation then we will not generate it.

Alternate solution for it is, since the generation of the image is happening locally, we can check locally and make sure image is generated properly before sending the PR and merging it.

So as a whole this PR is fine. I will create a follow-up issue as it might take some good amount of time because of the fact that the underlying package is not in our control and need to be studied hard to understand fully.

File: docs/Rakefile [hunk 2]
```diff
@@ -33,7 +33,7 @@ task :clean do
   rm_f ADOC_VARIABLES
   # find ditaa images based on the fact that they have a cache file
   DITAA_IMAGES = Rake::FileList.new do |list|
-    Rake::FileList.new("source/**/images/*.png.cache").each do |f|
+    Rake::FileList.new("source/**/images/*-template.png.cache").each do |f|
```
Comment (praveenkumar) [inline docs/Rakefile]: @budhrg `source/**/images/*-template.png.cache` what is the use of this, are you forgot to `cp` or `mv` to `minishift-architecture-template.png` or `minishift-architecture.png` ?

File: .gitignore [hunk 3]
```diff
@@ -43,7 +43,8 @@ docs/build
 docs/source/_tmp/*.md
 docs/source/command-ref/*.adoc
 docs/source/**/variables.adoc
-docs/source/using/images/minishift-architecture*
+docs/source/using/images/minishift-architecture.png.cache
+docs/source/using/images/minishift-architecture-template*
```
Comment (praveenkumar) [inline .gitignore]: :+1:

**Pull request merged**